executor/distsql.go

```diff
@@ -652,8 +664,10 @@ func (w *indexWorker) fetchHandles(ctx context.Context, result distsql.SelectRes
 }
 
 func (w *indexWorker) extractTaskHandles(ctx context.Context, chk *chunk.Chunk, idxResult distsql.SelectResult) (handles []int64, err error) {
-	handles = make([]int64, 0, w.batchSize)
-	for len(handles) < w.batchSize {
+	requiredNow := int(atomic.LoadInt64(&w.idxLookup.parentReqRows))
+	handles = make([]int64, 0, requiredNow)
+	for len(handles) < requiredNow {
```

Comment:
After this change, `tidb_index_lookup_size` can not control the batch size of a index lookup task?

Reply:
@qw4990 When we adjust the number of handles inside a task, I think we should also consider the value of `tidb_index_lookup_size`.

Reply:
How about set the `RequiredRows in indexWorker` to min(`tidb_index_lookup_size`, `RequiredRows from parent`)?
Then the `tidb_index_lookup_size` is able to control the number of handles fetched by `indexWorker` as before.